datacube/index/postgres/_datasets.py

```diff
@@ -736,29 +755,32 @@ def _do_search_by_product(self, query, return_fields=False, select_field_names=N
             if return_fields:
                 # if no fields specified, select all
                 if select_field_names is None:
-                    select_fields = tuple(field for name, field in dataset_fields.items()
-                                          if not field.affects_row_selection)
+                    select_fields = None
                 else:
-                    select_fields = tuple(dataset_fields[field_name]
-                                          for field_name in select_field_names)
+                    select_fields = tuple(
+                        dataset_fields[field_name]
+                        for field_name in select_field_names
+                        if field_name in dataset_fields  # and not dataset_fields[field_name].affects_row_selection
```

Comment:
Same comment as in postgis version, also looks like you forgot to uncomment the second half

Reply:
I think I forgot to remove it after I made it unnecessary.